library/pk.c

```diff
@@ -232,7 +240,11 @@ int mbedtls_pk_verify_restartable( mbedtls_pk_context *ctx,
                const unsigned char *sig, size_t sig_len,
                mbedtls_pk_restart_ctx *rs_ctx )
 {
-    if( ctx == NULL || ctx->pk_info == NULL ||
+    PK_VALIDATE_RET( ctx != NULL );
+    PK_VALIDATE_RET( hash != NULL );
+    PK_VALIDATE_RET( sig != NULL );
```

Comment:
We should require `( md_alg == MBEDTLS_MD_NONE && hash_len == 0 ) || hash != NULL`.

Reply:
Good point. Fixed and I added a corresponding test.